src/darwin/Framework/CHIP/MTRControllerAccessControl.mm

```diff
@@ -68,17 +69,23 @@ CHIP_ERROR Check(
     }
 };
 
-AccessControlDelegate gDelegate;
+struct ControllerAccessControl {
+    DeviceTypeResolver mDeviceTypeResolver;
+    AccessControlDelegate mDelegate;
+    ControllerAccessControl() { GetAccessControl().Init(&mDelegate, mDeviceTypeResolver); }
+};
+
+Global<ControllerAccessControl> gControllerAccessControl;
+
 } // anonymous namespace
 
+Global<DeviceTypeResolver> gDeviceTypeResolver;
+
 @implementation MTRControllerAccessControl
 
 + (void)init
 {
-    static dispatch_once_t onceToken;
-    dispatch_once(&onceToken, ^{
-        GetAccessControl().Init(&gDelegate, gDeviceTypeResolver);
-    });
+    (void) *gControllerAccessControl;
```

Comment:
The side-effect from `*` here is a bit weird.  It would be better if there were just an explicit `Emplace()` or `EnsureValue` or something method.

Reply:
Using `get()` instead of `operator*` now.